### DISCUSSION TITLE
LIMIT chunking method?

### DISCUSSION THREAD

**Discussion by midenok** [mydumper/mydumper#1376] [Ideas]
From [How mydumper splits a table?](https://github.com/mydumper/mydumper/wiki/How-mydumper-splits-a-table%3F) we can read about different chunking methods. AFAIU only file chunking and integer PK chunking may benefit from multithreading. And file chunks multithreading works only on myloader side while integer PK chunks mulithreading work on both mydumper and myloader (correct me if I'm wrong).

The question is, what if there were LIMIT chunking:
```sql
SELECT * FROM Orders LIMIT 10000 OFFSET 20000;
```
we would not need any other chunking methods to achieve multithreading as this one is simplest and most universal one (and it works on both mydumper/myloader). Do you know any drawbacks of this so they prevented implementation in mydumper?

**Comment by davidducos** [comment]
@midenok 
Do NOT implement it. There are performance penalties using LIMIT OFFSET, mostly on large values on offset.
https://planetscale.com/blog/mysql-pagination#performance-drawbacks-of-offset-limit-pagination
